### DISCUSSION TITLE
what is a sequential playlist suppose to be?

### DISCUSSION THREAD

**Discussion by matthewbrutonall** [AzuraCast/AzuraCast#4203] [General]
Hi fellow azuracasters!,

I have one of my playlists set to sequential, and for some reason it seems to be completely random still within the playlist.
I can see no method in it's choice of order. It is not in sequence by any of the other fields, like file size, playing time, etc. etc. 

I am trying to play an audio book with lots of small different tracks, so it is super important that it is played in strict order. There is no point in merging tracks either because I will not be playing it all in one go. I need it to pick up where it left off.

Any pointers ? Help greatly appreciated!

Matthew

**Comment by ncfernan** [comment]
@matthewbrutonall 

I use sequential playlist because I have an external shuffler program, so I need the songs to run in the order they are listed in that playlist.

You have to create a playlist (text file) with the sequence you want the files to play. Then upload the text file as a new playlist and set to sequential. The interpretation of "sequential" is "do not shuffle", but your playlist must be in the order you want the files to play.

Does your playlist have the files in the order you want them to play?

**Reply by matthewbrutonall** [reply]
Do you mean the "import from pls/m3u" option?
I created my playlist from within azuracast using the music library, and just selected all the files in a particular folder. But for some reason, even though the titles are in order in the folder, the playlist has them all random.

**Reply by ncfernan** [reply]
There is a subtle detail about sorted list of files: the Operating System (Linux, Windows) do not store the names of files in alphabetical order, they use the first slot available and put the filename in there. What you see in Azuracast is a list of files sorted by the program itself, they are not necessarily sorted in the directory (folder). When you use the "ls" shell command in Linux the "ls" program will sort the output no matter if the names are already in alphabetical order or not. The same occurs in Windows Explorer.

If you simply mark the directory to be included in the playlist, the filenames will be loaded in the order they are stored in the Operating System, which is almost random.

The procedure is what you wrote: create an empty playlist and mark it as sequential. Then import your file listing - with your particular sequence - into this newly created playlist (import from pls/m3u). That's it.

**Reply by matthewbrutonall** [reply]
Really appreciate all that. One last thing: I have my files organised in directories and I created playlists from there. If I import a playlist how do I tell it which directory the file is in? Does it matter?

**Reply by ncfernan** [reply]
I'll try to explain using screenshots.

This is the vision I have in the `music files` panel:

![image](https://user-images.githubusercontent.com/10008844/119494854-231f3c80-bd38-11eb-8054-22f1a1646e9e.png)

My song files are stored inside the `Tracks` directory:

![image](https://user-images.githubusercontent.com/10008844/119494997-4d70fa00-bd38-11eb-96a7-7bd4391a98b6.png)

As you can see, I have several subdirectories. One of them was selected to be inside my `default` playlist `Fbk`. The path you have to consider is:

![image](https://user-images.githubusercontent.com/10008844/119495435-c96b4200-bd38-11eb-9015-529ade26d24b.png)

This is the original directory tree I have in my notebook (Win10):

![image](https://user-images.githubusercontent.com/10008844/119495660-02a3b200-bd39-11eb-9f8b-09a70d93fc95.png)

This is the contents of my playlist file (M3U):

![image](https://user-images.githubusercontent.com/10008844/119495948-58785a00-bd39-11eb-8d29-d34b1c3a6504.png)

**Reply by matthewbrutonall** [reply]
ok, so a parallel directory. Cool.
I'll give that a go and see how I get on. 
Thanks so much!

**Comment by Vaalyn** [comment]
You need to manually sort the playlist either via importing a pls/m3u or by clicking on the `more` button on a playlist and selecting `Reorder`. There you can see the order of the playlist and with drag&drop or with the arrow buttons you can sort the playlist as you want.
